src/exampleTypes/ResultTable/TableBody.js

```diff
@@ -18,55 +19,71 @@ let TableBody = ({
-            <Row
-              key={getRowKey(x)}
-              record={getRecord(x)}
-              {...{ fields, visibleFields, hiddenFields }}
-            >
-              {_.map(
-                ({ field, display = x => x, Cell = Td }) => (
-                  <Cell
-                    key={field}
-                    className={field === stickyColumn ? 'sticky-column' : ''}
-                    style={{
-                      position: field === stickyColumn ? 'sticky' : '',
-                      left: field === stickyColumn ? 0 : '',
+    <>
+      <Tbody style={{ display: showLoader || showNoResults ? 'none' : '' }}>
+        {!!results.length &&
+          _.map(
+            x => (
+              <Row
+                key={getRowKey(x)}
+                record={getRecord(x)}
+                {...{ fields, visibleFields, hiddenFields }}
+              >
+                {_.map(
+                  ({ field, display = x => x, Cell = Td }) => (
+                    <Cell
+                      key={field}
+                      className={field === stickyColumn ? 'sticky-column' : ''}
+                      style={{
+                        position: field === stickyColumn ? 'sticky' : '',
+                        left: field === stickyColumn ? 0 : '',
+                      }}
+                    >
+                      {F.when(
+                        () => x.isBlank,
+                        blankResult,
+                        display
+                      )(_.get(field, getRecord(x)), getRecord(x))}
+                    </Cell>
+                  ),
+                  visibleFields
+                )}
+                {node.showOtherMatches && (
+                  <HighlightedColumn
+                    {...{
+                      node,
+                      additionalFields: _.result('additionalFields.slice', x),
+                      schema,
                     }}
-                  >
-                    {F.when(
-                      () => x.isBlank,
-                      blankResult,
-                      display
-                    )(_.get(field, getRecord(x)), getRecord(x))}
-                  </Cell>
-                ),
-                visibleFields
-              )}
-              {node.showOtherMatches && (
-                <HighlightedColumn
-                  {...{
-                    node,
-                    additionalFields: _.result('additionalFields.slice', x),
-                    schema,
-                  }}
-                />
-              )}
-            </Row>
-          ),
-          results
-        )}
-    </Tbody>
+                  />
+                )}
+              </Row>
+            ),
+            results
+          )}
+      </Tbody>
+      <Tbody style={{ display: showLoader || showNoResults ? '' : 'none' }}>
+        <Tr>
+          <Td colSpan={visibleFields.length} style={{ padding: 0 }}>
+            <Loader loading={showLoader}>{showLoader && IntroComponent}</Loader>
```

Comment:
Yeah that's it already was before, we were using the intro component to bump the height of the loading area. In all other cases, it would show no results.

Reply:
What was the original login behind the IntroComponent rendering?

Reply:
I thought the intro component was there before we get results for blank searches. @Akrion would know for sure

Reply:
Is there a way to detect when the search on the tree was not run yet?